src/arch/amdgpu/vega/insts/mubuf.cc

```diff
@@ -784,6 +793,25 @@ namespace VegaISA
     Inst_MUBUF__BUFFER_LOAD_USHORT::completeAcc(GPUDynInstPtr gpuDynInst)
     {
         if (instData.LDS) {
+            // LDS data is written as dword at minimum. Convert the type size
+            // in place in the dynamic instruction data pointer.
+            VecElemU32 *as_u32 =
+                reinterpret_cast<VecElemU32 *>(gpuDynInst->d_data);
+            VecElemU16 *as_u16 =
+                reinterpret_cast<VecElemU16 *>(gpuDynInst->d_data);
+
+            for (int lane = NumVecElemPerVecReg - 1; lane >= 0; --lane) {
+                // Reverse copy from largest to smallest index as the pointers
+                // are the same. Avoids clobbering.
+                if (gpuDynInst->exec_mask[lane]) {
+                    if (!oobMask[lane]) {
+                        as_u32[lane] = as_u16[lane];
```

Comment:
Does this ever go out of bounds?  Or is it just datatype, not array size, that differs?